x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/ContendedRegisterAnalyzeAction.java

```diff
@@ -156,7 +156,13 @@ public void onFailure(Exception e) {
         };
 
         if (request.getInitialRead() > request.getRequestCount()) {
-            blobContainer.getRegister(OperationPurpose.REPOSITORY_ANALYSIS, registerName, initialValueListener);
+            blobContainer.getRegister(OperationPurpose.REPOSITORY_ANALYSIS, registerName, initialValueListener.delegateFailure((l, r) -> {
+                if (r.isPresent()) {
+                    l.onResponse(r);
+                } else {
+                    l.onFailure(new IllegalStateException("register read failed due to contention"));
```

Comment:
@DaveCTurner raising your attention that it seems to fail `AzureSnapshotRepoTestKitIT/testRepositoryAnalysis` (see #108504)

Reply:
Thanks for the ping, clearly a bug: https://github.com/elastic/elasticsearch/pull/108900